ReSharper.FSharp/src/FSharp.Psi/src/CodeStyle/FSharpFormatterInfoProvider.cs

```diff
@@ -0,0 +1,245 @@
+using JetBrains.Application.Settings;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Impl.CodeStyle;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.CodeStyle
+{
+  [Language(typeof(FSharpLanguage))]
+  public class FSharpFormatterInfoProvider :
+    FormatterInfoProviderWithFluentApi<CodeFormattingContext, FSharpFormatSettingsKey>
+  {
+    public FSharpFormatterInfoProvider(ISettingsSchema settingsSchema) : base(settingsSchema)
+    {
+      lock (this)
```

Comment:
Is there an actual need for the lock?
Please look at how `Language` components are created, find usages of `LanguageAttribute` in the SDK/product code.

Reply:
Yes, there are asserts for locks in rules initialization.

Reply:
OK, there are asserts for this kind of locks in the base types, I see.